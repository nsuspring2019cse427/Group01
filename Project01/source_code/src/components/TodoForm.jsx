import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const FormInput = styled.input`
  width: 235px;
  outline: none;
  font-size: 13px;
  padding-top: 7px;
  padding-bottom: 7px;
  padding-left: 10px;
`;

const TodoForm = props => {
  const { handleSubmit, handleOnchangeInput, currentTodo } = props;

  return (
    <div>
      <p>Enter your todo and hit the Enter key </p>
      <form onSubmit={handleSubmit}>
        <FormInput
          placeholder="Enter new todo"
          onChange={handleOnchangeInput}
          value={currentTodo}
        />
      </form>
    </div>
  );
};

TodoForm.defaultProps = {
  currentTodo: "",
  handleSubmit: () => {},
  handleOnchangeInput: () => {}
};

TodoForm.propTypes = {
  currentTodo: PropTypes.string,
  handleSubmit: PropTypes.func,
  handleOnchangeInput: PropTypes.func
};

export default TodoForm;
