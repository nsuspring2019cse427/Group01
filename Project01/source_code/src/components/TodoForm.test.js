import React from 'react'
import Enzyme, {
    shallow
} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import TodoForm from './TodoForm'

describe('TodoForm Component', () => {
    beforeAll(() => {
        Enzyme.configure({
            adapter: new Adapter()
        })
    })

    it('renders correctly', () => {
        const component = shallow( < TodoForm / > )
        expect(component).toMatchSnapshot();
    })
})