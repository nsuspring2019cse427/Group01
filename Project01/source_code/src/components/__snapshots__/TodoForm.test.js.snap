exports[`TodoForm Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoForm
    currentTodo=""
    handleOnchangeInput={[Function]}
    handleSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Enter your todo and hit the Enter key 
        </p>,
        <form
          onSubmit={[Function]}
        >
          <styled.input
            onChange={[Function]}
            placeholder="Enter new todo"
            value=""
          />
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter your todo and hit the Enter key ",
        },
        "ref": null,
        "rendered": "Enter your todo and hit the Enter key ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <styled.input
            onChange={[Function]}
            placeholder="Enter new todo"
            value=""
          />,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Enter new todo",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Enter your todo and hit the Enter key 
          </p>,
          <form
            onSubmit={[Function]}
          >
            <styled.input
              onChange={[Function]}
              placeholder="Enter new todo"
              value=""
            />
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter your todo and hit the Enter key ",
          },
          "ref": null,
          "rendered": "Enter your todo and hit the Enter key ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <styled.input
              onChange={[Function]}
              placeholder="Enter new todo"
              value=""
            />,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Enter new todo",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;